sql/core/src/test/scala/org/apache/spark/sql/CollationSuite.scala

```diff
@@ -759,34 +761,38 @@ class CollationSuite extends DatasourceV2SQLBase with AdaptiveSparkPlanHelper {
   }
 
   test("hash based joins not allowed for non-binary collated strings") {
-    val in = (('a' to 'z') ++ ('A' to 'Z')).map(_.toString * 3).map(e => Row.apply(e, e))
+    spark.conf.set("spark.sql.codegen.wholeStage", false)
+    val in = (('a' to 'b') ++ ('A' to 'B')).map(_.toString).map(e => Row.apply(e, e))
```

Comment:
I assume that these are leftovers from some prototyping :)

Reply:
yes! overall I'd say this PR is still a prototype, so tests may stay messy until we converge on the correct approach